RPCServer/rpcwallet.py

```diff
@@ -199,31 +208,54 @@ def backup_wallet(self, afilename="bwallet.zip"):
         """
-        
-        # TODO - Help is welcome.
-        
         """
-        Wallet is a directory, referenced by self.path
-        each account is a json file, within a directory that is 2 chars long.
-        account abcdef is abcedf.json and lies in dir "ab"
-        
-        This dump has to parse all wallet json files (including "default.json" right in .wallet)
-        For each wallet, parse the "addresses" list and build the list of all addresses.
-        (ony Bismuth addresses, not the full priv/pub list)
-        
-        Please have a test setup and make sure code works before requesting PR.
+            Test possible path existence
         """
-        
-        print("TODO: Dump Wallet")
-        
-        # Should return a list of string or raise an exception
-        return ['TODO']        
+        dump_path = os.path.dirname(os.path.abspath(afilename))
 
+        if not os.path.exists(dump_path):
+            raise InvalidPath
 
+        if not os.path.exists(afilename):
+            if self.verbose:
+                print(afilename, "does not exist, creating")
+
+        with open(afilename, 'w') as outfile:
+            """ Write basic output """
+            outfile.write('# Wallet dump created by bismuthd ' + version + '\n')
+            outfile.write('# * Created on ' + datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ') + '\n')
+            if self.encrypted:
+                outfile.write('# * Wallet is Encrypted\n')
+            else:
+                outfile.write('# * Wallet is Unencrypted\n')
+
+            """ @TODO add block information like:
+                # * Best block at time of backup was 227221 (0000000026ede4c10594af8087748507fb06dcd30b8f4f48b9cc463cabc9d767),
+                #   mined on 2014-04-29T21:15:07Z
+            """
+            """ Walk all files, search for keys and parse them """
+            for root, dirs, files in os.walk(self.path):
+                for file in files:
+                    with io.open(os.path.join(root,file), 'r', encoding='utf-8-sig') as json_file:
+                        """ check if this is a json file """
+                        json_contents = json_file.read()
+                        if self._is_json(json_contents):
+                            res = json.loads(json_contents)
+                            account = os.path.splitext(file)[0]
+
+                            if 'addresses' in res:
+                                for address in res["addresses"]:
+                                    """
+                                        Output format:
+                                        privkey1 RESERVED account=account1 addr=address1
+                                        - RESERVED is for timestamp later
+                                    """
+                                    outfile.write(address[2] + ' RESERVED ' + 'account=' + account + ' addr=' + address[0] + '\n')
+        return True
```

Comment:
bitcoind impl. return null in every case (success or failure), only message changes.
So I'd return None here.